core/src/main/scala/kafka/server/ReplicaFetcherThread.scala

```diff
@@ -161,15 +157,6 @@ class ReplicaFetcherThread(name: String,
     }
   }
 
-  private def maybeWarnIfOversizedRecords(records: MemoryRecords, topicPartition: TopicPartition): Unit = {
-    // oversized messages don't cause replication to fail from fetch request version 3 (KIP-74)
-    if (metadataVersionSupplier().fetchRequestVersion <= 2 && records.sizeInBytes > 0 && records.validBytes <= 0)
```

Comment:
It seems another use case of `fetchRequestVersion` is dead code. Maybe we can remove it as well? 
```scala
      val version: Short = if (metadataVersion.fetchRequestVersion >= 13 && !fetchData.canUseTopicIds) {
        12
      } else {
        metadataVersion.fetchRequestVersion
      }
```


https://github.com/apache/kafka/blob/5982943267ecbc53c42e80feb2d8506aad0c0559/core/src/main/scala/kafka/server/RemoteLeaderEndPoint.scala#L206

Reply:
I think you're saying that IBP 3.3 implies fetch request 13 or higher. That makes sense, I pushed a commit that removes that check.